### PR TITLE
add reading of local .svnignore file over defaults

### PR DESCRIPTION
**Pull request by afragen**
If local .svnignore file exists use it's contents for setting `svn propset svn:ignore ...` in lieu of defaults.

This should obviate #36 and provide a more versatile alternative to #11

### CODE REVIEW

**Comment by afragen** [comment]
I'm still working out some kinks.

**Comment by afragen** [comment]
It seems setting `svn propset svn:ignore ...` is flaky and doesn't work all the time. This commit now uses a local `.svnignore` file with `svn del` to remove the files from svn prior to commit.

Seems to work here in limited testing.

**Comment by afragen** [comment]
I think the kinks have been worked out. It’s been working fine for me.

File: deploy.sh [hunk 1]
```diff
@@ -157,12 +157,22 @@ svn checkout $SVNURL $SVNPATH --depth immediates
 svn update --quiet $SVNPATH/trunk --set-depth infinity
 
 echo "Ignoring GitHub specific files"
-svn propset svn:ignore "README.md
+# Use local .plugins_svnignore if present
+if [ -f ".plugins_svnignore" ]; then
+	echo "Using local .plugins_svnignore"
+	SVNIGNORE=$( <.pluginsvnignore )
```
Comment (GaryJones) [inline deploy.sh]: This file name doesn't seem consistent (missing underscore) with everything else.

File: README.md [hunk 2]
```diff
@@ -33,9 +33,17 @@ These are the steps that the script takes:
 2. Ensure that the shell script is executable. In Mac / Unix, run `chmod +x deploy.sh`.
 3. Run the script with `sh deploy.sh`. You can also double-click it in Finder / Explorer to start it.
 4. You'll now be guided through a set of questions.
- 
+
 I prefer to keep this script in the root of my projects directory. Each project directory is named as the plugin slug, as is the corresponding GitHub repo. To use, just call the script, enter the plugin slug, confirm or amend default suggestions, and sit back as the code is sent to SVN and git repos including tags. The commit messages here are hard-coded for consistency.
 
+## Extras
+
+You may define your own `.plugins_svnignore` file similar to an `svn propset svn:ignore .plugins_svnignore` command. This will remove any listed files and/or directories from committing to the Plugins Directory.
```
Comment (GaryJones) [inline README.md]: Why is the filename `.plugins_svnignore`? What's the origin of that name?

Comment (afragen) [reply README.md]: I made it up. I initially had `.svnignore` but added `plugins_` for Plugins Directory SVN.

File: deploy.sh (hunk 1 above)
Comment (afragen) [reply deploy.sh]: Must be a typo on my part when I changed from `.svnignore` to `.plugins_svnignore`

Comment (afragen) [reply deploy.sh]: I probably should just go with my first choice `.svnignore`

File: README.md (hunk 2 above)
Comment (afragen) [reply README.md]: I probably should just go with my first choice,  `.svnignore`

File: deploy.sh (hunk 1 above)
Comment (afragen) [reply deploy.sh]: Changed back to `.svnignore`

**Comment by afragen** [comment]
Back to using `.svnignore`

**Pull request merged**